custom_components/dbuezas_eq3btsmart/climate.py

```diff
@@ -165,19 +165,10 @@ def __init__(self, _mac: str, _device_name: str, _hass: HomeAssistant):
 
     async def async_added_to_hass(self) -> None:
         _LOGGER.debug("[%s] adding", self._device_name)
-
-        # Small hack: HA might call async_update before the entity has been fully added.
-        # In this case the first update callback will be missed and _is_available won't be set.
-        # We circumvent this by checking whether the mode of the thermostat has been set.
-        # If this is the case the first update must have been sucessful.
-        if self._thermostat.mode != eq3.Mode.Unknown:
-            self._on_updated()
-
-        self._thermostat.on_update = self._on_updated
+        self.schedule_update_ha_state(force_refresh=True)
```

Comment:
I think this should be async_schedule_update_ha_state so that this doesn't block until the update is done.

Reply:
Oh, you are right.